### ISSUE TITLE
[List]: Screen reader does not announce list items as clickable

### ISSUE THREAD

**Issue by wilson-xu-sap** [SAP/ui5-webcomponents#10914]
### Describe the bug

Screen reader users are not informed that the list items are clickable or active, leading to a lack of clarity about the interactive nature of the records.

### Isolated Example

https://sap.github.io/ui5-webcomponents-react/v2/?path=/docs/data-display-list--docs

### Reproduction steps

Enable screen reader
Focus on list item and use arrow keys to switch list item.

### Expected Behaviour

Screen readers should announce each list item as clickable or active

### UI5 Web Components for React Version

2.7

### UI5 Web Components Version

2.7

### Browser

Chrome, Edge

### Relevant log output

```shell

```

### Organization

SF

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by Todor-ads** [comment]
Hello @SAP/ui5-webcomponents-topic-p  ,

could you please check this issue?
The issue can be reproduced using the provided link.

Thanks,
Todor
